scripts/e2e.ts

```diff
@@ -8,8 +8,20 @@ import childProcess from 'child_process';
 import events from 'events';
 import util from 'util';
 
+/**
+ * `isDevEnv` is executed immediately via the `buildUtils` import, so we need to
+ * assign environment variables as early as possible. Otherwise, environment
+ * variables like `process.env.NODE_ENV` and `process.env.RD_TEST` will be
+ * `undefined`.
+ */
+process.env.NODE_ENV = 'production';
+process.env.RD_TEST = 'e2e';
+process.env.RD_ENV_EXTENSIONS = '1';
+
+// eslint-disable-next-line import/first
```

Comment:
I'm open to alternative approaches, but this appeared to be most straightforward at the moment.

Reply:
Should `NODE_ENV` be `production` here? We run e2e tests against `yarn dev`, right?

How do you feel about dropping the dependency from `build-utils` to `@pkg/utils/environment`, and instead making `buildUtils.isDevelopment` settable (and making it `true` except being set to `false` from `build`/`package`/etc.)?

Reply:
> Should NODE_ENV be production here? We run e2e tests against yarn dev, right?

Kind of.. we perform "production" builds of the frontend by invoking `node_modules/nuxt/bin/nuxt.js` [^1] in `scripts/e2e.ts` and then do development-optimized builds for preload and main. This behavior is the root of mismatched/unexpected values for `NODE_ENV` that webpack 5 will complain about. This would be much more straightforward if we just did a `yarn dev`. 

> How do you feel about dropping the dependency from build-utils to @pkg/utils/environment, and instead making buildUtils.isDevelopment settable (and making it true except being set to false from build/package/etc.)?

I think I like this approach better, it has less potential for mixing up the meaning of `NODE_ENV`.  

[^1]: We originally passed in the value of `test`, so maybe it was never quite right in the first place